wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory+UIAlertController.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+extension SettingsPropertyFactory {
+
+    typealias AlertLocale = L10n.Localizable.Self.Settings.PrivacyAnalytics.Alert
+
+    func showAnalyticsConsentAlert(completion: @escaping (Bool) -> Void) {
+        let alertController = UIAlertController(
+            title: AlertLocale.title,
+            message: AlertLocale.message,
+            preferredStyle: .alert
+        )
+
+        let actions: [(title: String, style: UIAlertAction.Style, handler: (UIAlertAction) -> Void)] = [
+            (AlertLocale.Button.accept, .default, { _ in completion(true) }),
+            (AlertLocale.Button.reject, .cancel, { _ in completion(false) }),
+            (AlertLocale.Button.privacyPolicy, .default, { [weak self] _ in
+                self?.presentPrivacyPolicy()
+                completion(false)
+            })
+        ]
+
+        actions.forEach { action in
+            alertController.addAction(UIAlertAction(title: action.title, style: action.style, handler: action.handler))
+        }
+
+        UIApplication.shared.topmostViewController(onlyFullScreen: false)?
```

Comment:
If there is no `topmostViewController` this would be a serious issue. I think you should guard let for it and log an error if it is not there.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/cd5db3f9625e59c227722780a71e1e219e12d4aa